net: app: Bind to correct address family

jukkar opened the pull request:
Use the remote address to determine what address family to bind.
This prevents extra context to be created.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
net: app: Bind to correct address family

Use the remote address to determine what address family to bind.
This prevents extra context to be created. In order to avoid
parsing the peer address string multiple times, the client
init function is re-factored.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

pfalcon commented on subsys/net/lib/app/client.c:
```diff
@@ -322,7 +322,20 @@ int net_app_init_client(struct net_app_ctx *ctx,
 		memcpy(&addr, client_addr, sizeof(addr));
 	} else {
-		addr.sa_family = AF_UNSPEC;
+		if (peer_addr_str) {
```
There's also "peer_addr" arg to this function (net_app_init_client), shouldn't it be consulted to avoid extra name resolution run?

Also, later we'd need to resolve peer_addr_str anyway, so make sense to do that once, though looking at the code, that would require heavier refactoring of the code...

jukkar committed:
net: app: Close context even if there is no connection

If there is no connection to server, then _net_app_select_net_ctx()
will return NULL. This is perfectly fine and we can just continue.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
tests: net: app: Simple testcases for net-app testing

These are simple tests for net-app testing. These do not try to
actually send anything to net but test mainly net-app setup and
close functionality.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar commented:
New version uploaded, I also added simple unit tests that test the refactored client APIs.

pfalcon reviewed (approved):
Thanks for taking care to refactor remote addr resolving!

Few nRF51 board tests fail due to RAM overflow though, I guess this is not directly related to the refactor per se, so just need do be excluded based on low RAM.
